internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -389,3 +394,18 @@ func (n *Networkd) decideHostname() (hostname string, domainname string, address
 	// Only return the hostname portion of the name ( strip domain bits off )
 	return hostname, domainname, address, nil
 }
+
+// Ready exposes the readiness state of networkd.
+func (n *Networkd) Ready() bool {
+	n.readyLock.Lock()
+	defer n.readyLock.Unlock()
+
+	return n.ready
+}
+
+// SetReady sets the readiness state of networkd.
+func (n *Networkd) SetReady() {
+	n.readyLock.Lock()
+	n.ready = true
+	n.readyLock.Unlock()
```

Comment:
nit: The `Lock()` and `defer Unlock()` strategy is used in `Ready`, but not here. That is preferred, but either way the strategy/method (whatever you want to cal it) should align/match.